identity-iota/src/credential/validator.rs

```diff
@@ -19,28 +30,29 @@ use crate::error::Result;
 use crate::tangle::Client;
 use crate::tangle::TangleResolve;
 
-#[derive(Clone, Debug, PartialEq, Serialize)]
-pub struct CredentialValidation<T = Object> {
-  pub credential: Credential<T>,
-  pub issuer: DocumentValidation,
-  pub subjects: BTreeMap<String, DocumentValidation>,
-  pub verified: bool,
+// used internally during credential validation
+#[allow(clippy::large_enum_variant)] // The smaller variants occur at most once each
+enum CredentialDeficiencyEvent<'a> {
+  // the Url corresponds to the url of the deactivated document
+  DeactivatedSubjectDocument((&'a Url, DocumentValidation<Deactivated>)),
+  Expired(Timestamp),
+  Dormant(Timestamp),
 }
-
-#[derive(Clone, Debug, PartialEq, Serialize)]
-pub struct PresentationValidation<T = Object, U = Object> {
-  pub presentation: Presentation<T, U>,
-  pub holder: DocumentValidation,
-  pub credentials: Vec<CredentialValidation<U>>,
-  pub verified: bool,
+impl CredentialDeficiencyEvent<'_> {
+  fn associated_category(&self) -> CredentialDeficiency {
```

Comment:
```suggestion
  fn as_deficiency(&self) -> CredentialDeficiency {
```
Or `to_deficiency`? Not sure.